### PR TITLE
v2.0 Add unit tests for `as_tibble` and `as.data.frame`

### PR DESCRIPTION
**Pull request by edelarua**
**What changes are proposed in this pull request?**

Add v2.0 unit tests for `as_tibble` and `as.data.frame`. 

Part of https://github.com/insightsengineering/sme-tasks/issues/689

Closes https://github.com/insightsengineering/sme-tasks/issues/686

--------------------------------------------------------------------------------

Reviewer Checklist (if item does not apply, mark is as complete)

- [ ] Ensure all package dependencies are installed by running `renv::install()`
- [ ] PR branch has pulled the most recent updates from master branch. Ensure the pull request branch and your local version match and both have the latest updates from the master branch.
- [ ] If an update was made to `tbl_summary()`, was the same change implemented for `tbl_svysummary()`?
- [ ] If a new function was added, function included in `_pkgdown.yml`
- [ ] If a bug was fixed, a unit test was added for the bug check
- [ ] Run `pkgdown::build_site()`. Check the R console for errors, and review the rendered website.
- [ ] Code coverage is suitable for any new functions/features. Review coverage with `withr::with_envvar(new = c("NOT_CRAN" = "true"), covr::report())`. Begin in a fresh R session without any packages loaded. 
- [ ] R CMD Check runs without errors, warnings, and notes
- [ ] `usethis::use_spell_check()` runs with no spelling errors in documentation

When the branch is ready to be merged into master:
- [ ] Update `NEWS.md` with the changes from this pull request under the heading "`# gtsummary (development version)`". If there is an issue associated with the pull request, reference it in parentheses at the end update (see `NEWS.md` for examples).
- [ ] Increment the version number using `usethis::use_version(which = "dev")` 
- [ ] Run `codemetar::write_codemeta()`
- [ ] Run `usethis::use_spell_check()` again
- [ ] Approve Pull Request
- [ ] Merge the PR. Please use "Squash and merge".

### CODE REVIEW

File: tests/testthat/test-as_tibble.R [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+  expect_snapshot(res %>% as.data.frame())
+
+  expect_silent(res <- as_tibble(t3))
+  expect_snapshot(res %>% as.data.frame())
+})
+
+test_that("as_tibble(col_labels=) works", {
+  expect_equal(
+    as_tibble(t2, col_labels = FALSE) %>% names(),
+    c("label", "estimate", "conf.low", "p.value")
```
Comment (edelarua) [inline tests/testthat/test-as_tibble.R:30]: @ddsjoberg is this label expected to be `"conf.low"` here (not `"ci"`)?

**Comment by ddsjoberg** [comment]
Thank you so much for this @edelarua !!

> Tests for column merging currently fail unless run on branch v2.0-tbl_merge.

Just a couple of minutes ago, I merged a PR that make it possible to merge the `tbl_merge()` branch. I'll do that first, and update this PR later this evening (if everything goes as planned)

File: tests/testthat/test-as_tibble.R (hunk 1 above)
Comment (ddsjoberg) [reply tests/testthat/test-as_tibble.R:30]: That is correct. When the package was originally written gt didn't have column merging capabilities, and we'd manually create a new `ci` column. But we don't need to do that any more

Comment (edelarua) [reply tests/testthat/test-as_tibble.R:30]: Sounds good! Thanks for clarifying

**Comment by edelarua** [comment]
@ddsjoberg this PR should be ready for review

**Comment by ddsjoberg** [comment]
Can you add a test with grouping columns? Like this?

```r
tbl <- tbl_summary(trial, include = age, missing = "no")

tbl_stack(list(tbl, tbl), group_header = c("T1", "T2")) |> 
  as_tibble()
```

**Comment by edelarua** [comment]
> Can you add a test with grouping columns?

@ddsjoberg a test has been added!

File: tests/testthat/_snaps/as_tibble.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
+# as_tibble works with standard use
```
Comment (ddsjoberg) [inline tests/testthat/_snaps/as_tibble.md:1]: This is a good test. Can we make the table shorter, so if it ever changes in the future it's easier to see the differences? Maybe one continuous variable, one dichotomous, and one categorical?

File: tests/testthat/_snaps/as_tibble.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+      as.data.frame(res)
+    Output
+        **Characteristic** **N** **OR** **95% CI** **p-value**
+      1                Age   183   1.02 1.00, 1.04        0.10
+      2              Grade   193   <NA>       <NA>        <NA>
+      3                  I  <NA>   <NA>       <NA>        <NA>
+      4                 II  <NA>   0.95 0.45, 2.00         0.9
+      5                III  <NA>   1.10 0.52, 2.29         0.8
+
+# as_tibble works with bold/italics
```
Comment (ddsjoberg) [inline tests/testthat/_snaps/as_tibble.md:52]: Same comment as above. I think we can do the same test but with a smaller table.

File: tests/testthat/_snaps/as_tibble.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+      as.data.frame(res)
+    Output
+        **Characteristic**    **N** **OR**  **95% CI** **p-value**
+      1              _Age_   183.00   1.02 0.997, 1.04        0.10
+      2            _Grade_ 193.0000   <NA>        <NA>        <NA>
+      3              __I__     <NA>   <NA>        <NA>        <NA>
+      4             __II__     <NA>   0.95 0.446, 2.00         0.9
+      5            __III__     <NA>   1.10 0.524, 2.29         0.8
+
+# as_tibble works with column merging
```
Comment (ddsjoberg) [inline tests/testthat/_snaps/as_tibble.md:127]: can we update this heading to be more clear? this test has both column merging for the confidence interval, and table merging.

File: tests/testthat/test-as_tibble.R [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+  tbl_regression()
+
+tbl <- tbl_merge(list(t4, t5))
+
+test_that("as_tibble works with column merging", {
+  expect_silent(res <- as_tibble(tbl))
+  expect_snapshot(res |> as.data.frame())
+})
+
+test_that("as_tibble(fmt_missing=) works", {
```
Comment (ddsjoberg) [inline tests/testthat/test-as_tibble.R:152]: Can you also add a test showing that missing values are not formatted when FALSE?

File: tests/testthat/test-as_tibble.R [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+  bold_labels() |>
+  italicize_levels()
+
+t2 <- t2 |>
+  bold_labels() |>
+  italicize_levels()
+
+t3 <- t3 |>
+  bold_levels() |>
+  italicize_labels()
```
Comment (ddsjoberg) [inline tests/testthat/test-as_tibble.R:49]: Can we discuss a unified way of having objects outside teh code chunk to keep it organized across all the testing files?

Should we always put these types of objects at the top of the script? Something else?

File: tests/testthat/test-as_tibble.R [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+t2 <- glm(response ~ age + grade, trial, family = binomial(link = "logit")) |>
+  tbl_regression(exponentiate = TRUE)
+
+t3 <- tbl_uvregression(
+  trial |> dplyr::select(response, age, grade),
+  method = glm,
+  y = response,
+  method.args = list(family = binomial),
+  exponentiate = TRUE
+)
```
Comment (ddsjoberg) [inline tests/testthat/test-as_tibble.R:13]: for these objects we use again and again, can we please name them in a way that helps us identify what kind of table they are, e.g. `t1_summary`, `t2_regression`?

**Review by ddsjoberg — CHANGES_REQUESTED** [review]
Just a few cosmetic changes, then this is good to go. Thank you!!

**Review by ddsjoberg — APPROVED** [review]
This is great, thank you !!

**Pull request merged**